### PR TITLE
added support for compiling without module option

### PR DESCRIPTION
**Pull request by Linowitch**
Fix for the following issue.

Argument module cannot be ""
#211 opened 7 days ago by Linowitch

### CODE REVIEW

**Comment by basarat** [comment]
Awesome. Would prefer if we warned the user of a bad `module` option instead of silently only taking valid values.

**Comment by Linowitch** [comment]
I've just added the warnings.

File: tasks/modules/compile.js [hunk 1]
```diff
@@ -148,7 +148,17 @@ function compileAllFiles(targetFiles, target, task, targetName) {
     }
     // string options
     args.push('--target', task.target.toUpperCase());
-    args.push('--module', task.module.toLowerCase());
+    // check the module compile option
+    var moduleOptionString = task.module.toLowerCase();
```
Comment (basarat) [inline tasks/modules/compile.js]: Wouldn't call '' a valid value. Also I've seen friends pass in null e.g https://github.com/TypeStrong/atom-typescript/issues/123#issuecomment-76680705 Which will break lower case call 

If they don't want to specify a module they should remove the option.

**Comment by Linowitch** [comment]
Ok, then we have to change the documentation as well.

I think we have two options then.

1. "amd" stays as default, to remove module it needs to set to "false" or "null".
   forces the user to set the module explicit to "null" or "false"

2, the default will be changed to "null" and if set only "amd" or "commonjs" are allowed.
   in this case the user needs only to ommit the module option. Like you described in the issue 123.

Which way shall we go?

**Comment by basarat** [comment]
Recommendation to update the docs: "default to be nothing, If you want to set it you set it to either amd or commonjs" 

I don't see the motivation of specifically *clearing* the option.

**Comment by Linowitch** [comment]
Did the changes and adjusted the documentation.

**Comment by basarat** [comment]
Awesome ❤️😘

**Pull request merged**